### PR TITLE
[DROOLS-3730] Add support to DMN constraints on inputs (report input validation errors)

### PR DESCRIPTION
**Pull request by danielezonca**
@kkufova @gitgabrio @Rikkola 

How to test:
- Import a dmn file with an input constraint like [DMNString.dmn.txt](https://github.com/kiegroup/drools-wb/files/3005478/DMNString.dmn.txt) (remove `.txt`)
- Create a scesim file on it
- Put an invalid input (accepted values are only `"Example1"`, `"Example2"` and `"Example3"`)
- Run

https://issues.jboss.org/browse/DROOLS-3730

### CODE REVIEW

**Comment by danielezonca** [comment]
jenkins retest this please

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/pom.xml [hunk 1]
```diff
@@ -136,6 +136,16 @@
       <artifactId>jackson-databind</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>org.kie</groupId>
+      <artifactId>kie-dmn-core</artifactId>
+    </dependency>
+
+    <dependency>
```
Comment (gitgabrio) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/pom.xml]: @danielezonca 
I know it is unavoidable, but did you take a look at relationship/binding this dependency creates with other jars? I mean, with jars taken in by kie-dmn-core/kie-dmn-model

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/runner/DMNScenarioRunnerHelper.java [hunk 2]
```diff
@@ -90,6 +97,34 @@ public void verifyConditions(SimulationDescriptor simulationDescriptor,
...
+    protected Stream<String> retrieveInputNodeName(DMNMessage message) {
+        Object sourceReference = message.getSourceReference();
+        if (sourceReference instanceof NamedElement &&
+                ((NamedElement) sourceReference).getName() != null) {
+            return Stream.of(((NamedElement) sourceReference).getName());
+        }
+        return Stream.empty();
+    }
+
+    protected void applyErrorIfUsed(String name, List<ScenarioGiven> givens) {
```
Comment (gitgabrio) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/runner/DMNScenarioRunnerHelper.java]: @danielezonca 
Could you add two lines of javadoc to this method to explain its scope? The previous twos are pretty easy to understand by name, this one not.

Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/main/java/org/drools/workbench/screens/scenariosimulation/backend/server/runner/DMNScenarioRunnerHelper.java]: Done

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/pom.xml (hunk 1 above)
Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/pom.xml]: Yes this dependency binds dmn modules and scenario more that _expected_ but it cannot be avoided. Maybe in the future can be removed or at least limited to `test` scope

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/test/java/org/drools/workbench/screens/scenariosimulation/backend/server/runner/DMNScenarioRunnerHelperTest.java [hunk 3]
```diff
@@ -192,4 +204,58 @@ public void createObject() {
...
+        List<String> result = runnerHelper.retrieveInputNodeName(dmnMessageMock).collect(toList());
+        assertEquals(1, result.size());
+        assertEquals(INPUT_NAME, result.get(0));
+
+        result = runnerHelper.retrieveInputNodeName(mock(DMNMessage.class)).collect(toList());
+        assertEquals(0, result.size());
+    }
+
+    @Test
+    public void applyErrorIfUsed() {
```
Comment (kkufova) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/test/java/org/drools/workbench/screens/scenariosimulation/backend/server/runner/DMNScenarioRunnerHelperTest.java]: This should probably be renamed too.

Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-backend/src/test/java/org/drools/workbench/screens/scenariosimulation/backend/server/runner/DMNScenarioRunnerHelperTest.java]: Done

**Comment by danielezonca** [comment]
@manstis 

Can you please squash and merge this?

**Pull request merged**